update.sh

```diff
@@ -45,7 +45,8 @@ for version in "${versions[@]}"; do
 		echo >&2 "warning: cannot determine sha for $version (tried all of ${allVersions[*]}); skipping"
 		continue
 	fi
-	shaVal="$(echo "$shaPage" | sed -r "s/.*Ruby ${fullVersion}<\/a><br \/>\s*sha256: ([^<]+).*/\1/")"
+	versionReleasePage="$(curl -fsSL 'https://www.ruby-lang.org/en/downloads/releases/' | grep "<td>Ruby $fullVersion</td>" -A 2 | awk -F'"' '{print $2}' | xargs)"
```

Comment:
The reason why I didn't touch it it's just because we use it before switching to xz, which means we use it to determine the versions even we just download gz tarballs, so IMO it's not related to this PR, but will like to send another PR to update this part.